src/test/java/uk/gov/pay/api/it/CreatePaymentIT.java

```diff
@@ -111,6 +113,61 @@ public void createCardPaymentWithMetadata() {
         connectorMockClient.verifyCreateChargeConnectorRequest(GATEWAY_ACCOUNT_ID, createChargeRequestParams);
     }
 
+    @Test
+    public void createAChargeWithMetadataAgreementIdAndSaveAgreement() {
+        publicAuthMockClient.mapBearerTokenToAccountId(API_KEY, GATEWAY_ACCOUNT_ID, CARD);
+
+        CreateChargeRequestParams createChargeRequestParams = aCreateChargeRequestParams()
+                .withAmount(100)
+                .withDescription(DESCRIPTION)
+                .withReference(REFERENCE)
+                .withReturnUrl(RETURN_URL)
+                .withAgreementId("12345678901234567890123456")
+                .withSavePaymentInstrumentToAgreement(true)
+                .withMetadata(Map.of("reconciled", true, "ledger_code", 123, "fuh", "fuh you"))
```

Comment:
We don’t really care about metadata stuff here, so kill it. What we want is something like `createCardPayment` with `.body("save_payment_instrument_to_agreement", is(true))` and `.body("agreement_id", is(whatever))`.